src/dice/add_dice/randomizer.rb

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+class AddDice
+  class Randomizer
+    attr_reader :dicebot, :cmp_op, :dice_list, :sides
+
+    def initialize(bcdice, dicebot, cmp_op)
+      @bcdice = bcdice
+      @dicebot = dicebot
+      @cmp_op = cmp_op
+      @sides = 0
+      @dice_list = []
+    end
+
+    def roll(times, sides, critical)
+      total = 0
+      results_list = []
+
+      loop_count = 0
+      queue = [times]
+      while !queue.empty?
+        times = queue.shift
+        val, dice_list = roll_once(times, sides)
+
+        total += val
+        results_list.push(dice_list)
+
+        enqueue_reroll(dice_list, queue, times)
+        @dicebot.check2dCritical(critical, val, queue, loop_count)
+        loop_count += 1
+      end
+
+      total = @dicebot.changeDiceValueByDiceText(total, results_list.flatten, @cmp_op, sides)
+
+      text = total.to_s
+      results_list.each do |list|
+        text += '[' + list.join(',') + ']'
+      end
+
+      [total, text]
+    end
+
+    private
+
+    def roll_once(times, sides)
+      @sides = sides if @sides < sides
+
+      if sides == 66
+        return rollD66(dice_wk)
+      end
+
+      _, dice_list, = @bcdice.roll(times, sides, @dicebot.sortType & 1)
+      dice_list = dice_list.split(",").map(&:to_i)
+      @dice_list.concat(dice_list)
+
+      total = dice_list.inject(&:+) || 0
```

Comment:
```suggestion
      total = dice_list.inject(0, &:+)
```

`Enumerable#inject` では、初期値を指定しておくと、より安全です。1.8.7でも使えます。
参考：https://docs.ruby-lang.org/ja/1.8.7/class/Enumerable.html#I_INJECT